commit 6f8f95a447a2e4ea8c30601bb67825267e81bcbc (refs/changes/82/782/1)
Author: Richard Bradley <richard.bradley@joyent.com>
Date:   2016-10-26T11:53:15+01:00 (2 years, 11 months ago)
    
    MANTA-2799 "manta-adm update" should fail when given non-existent service

diff --git a/lib/adm.js b/lib/adm.js
index 94e2b1a..fce45db 100644
--- a/lib/adm.js
+++ b/lib/adm.js
@@ -1383,6 +1383,11 @@ maAdm.prototype.generatePlan = function (callback, service, noreprovision)
 	assert.ok(this.ma_instances_wanted !== null);
 	assert.ok(this.ma_plan === null);
 
+	if(service && !svcs.serviceNameIsValid(service)){
+		callback(new VError('unrecognized service: "%s"', service));
+		return;
+	}
+
 	var cnid, dcnconf, svcconfig, svcid, svcname, actual;
 	var svcname2uuids = {};
 	var usedany = false;
